rpcs3/Emu/Cell/lv2/sys_timer.cpp

```diff
@@ -290,6 +290,14 @@ error_code sys_timer_sleep(ppu_thread& ppu, u32 sleep_time)
 
 error_code sys_timer_usleep(ppu_thread& ppu, u64 sleep_time)
 {
+	if (sleep_time < 10) {
+		u64 start = get_system_time();
+		while ((get_system_time() - start) < sleep_time) {
```

Comment:
1. This implementation will pressure get_system_time() too greatly to spin on it with no pause (_mm_pause) will be extremely short. Consider calling busy_wait() with a number of cycles to wait, it uses rdtsc and no syscalls so it should be responsive.
```
while (get_system_time() - start , sleep_time)
    busy_wait(4000);
```
4000 cycles should take around 1us at 4ghz and would be better imo than an infinite loop with no reprieve

2. According to some unix and posix spec documents, usleep(0) should return immediately and is actually a NOP. In fact this may be the reason this has any impact at all. According to https://lists.freebsd.org/pipermail/freebsd-arch/2012-March/012416.html no system has accurate usleep/nanosleep implementation, I suspect for simplicity the min timeout if greater than 1 is something around 100us but I would have to test this on real hardware.

These findings may have other consequences, e.g for thread_ctrl::wait_for called with very small timeout values. A separate implementation may be required for them or this could be handled by the callers.
I'll leave the review of the implications, should they exist, up to neko, i.e whether other functions such as wait_for and sleep_timeout need updating as well, or even imp_wait.

Reply:
usleep(0) may be NOP but it's different syscall. This syscall still may cause context switch in the presence of other threads with similar priority (like yield), unless there is strong evidence it doesn't.